### PR TITLE
Extend Apply verb to extensions, Bump to K8s 1.22.

### PR DESCRIPTION
**Pull request by mattmoor**
This was motivated by the linked issue, and does the work to upgrade PKG to 1.22 libs.

Fixes: https://github.com/knative/pkg/issues/2320

/kind enhancement
/hold

HOLDing this until we're clear that we *should* be bumping to 1.22 libs

**Release Note**

```release-note
Bump to K8s 1.22 libs
```

**Docs**
```docs

```

### CODE REVIEW

**Comment by antoineco** [comment]
@mattmoor I didn't look into the details, but is anything stopping us from updating the code generator (add the `Apply` method) without actually pulling 1.22 dependencies into Knative yet?

This would allow an upgrade in two steps, and downstream could update their generated code ahead of time while upstream takes the time to do what's necessary.

Prometheus [recently bumped their dependencies](https://github.com/prometheus/prometheus/blob/v2.31.1/go.mod#L74-L76), which means it is now difficult to depend both on Knative and (up-to-date) Prometheus.

**Comment by mattmoor** [comment]
I paged out a lot of the context here, but I think there was something (new type?).  You are welcome to try splitting pieces off of this.

**Comment by antoineco** [comment]
Right, the new `ApplyScale()` method has the following [signature](https://github.com/kubernetes/kubernetes/blob/v1.22.3/staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/deployment.go#L59):

```go
// import applyconfigurationsautoscalingv1 "k8s.io/client-go/applyconfigurations/autoscaling/v1"

ApplyScale(
	ctx context.Context,
	objectName string,
	scale *applyconfigurationsautoscalingv1.ScaleApplyConfiguration,
	opts metav1.ApplyOptions,
) (*autoscalingv1.Scale, error)
```

The type [`applyconfigurationsautoscalingv1.ScaleApplyConfiguration`](https://github.com/kubernetes/kubernetes/blob/v1.22.3/staging/src/k8s.io/client-go/applyconfigurations/autoscaling/v1/scale.go#L27-L34) was first introduced in Kubernetes `v1.22.0` 🧐 

So no choice but to update the libs.

Do you have cycles to rebase or would you prefer me to take over in a separate PR?

**Comment by mattmoor** [comment]
I'm happy to rebase if we're 👍  on bumping, I just wanted to minimize that number if it was going to be a while.

**Comment by antoineco** [comment]
@markusthoemmes based on your experience bumping k8s from 1.20 to 1.21, do you remember any specific issue downstream in the process?

**Comment by markusthoemmes** [comment]
Not that I recall. Once we bump here, we're fine downstream usually.

**Comment by nak3** [comment]
1.1 was released. @dprotaso Is this ready to go now?

**Comment by mattmoor** [comment]
I can try to dust off this PR over the weekend if we are cool with the bump next release.

**Comment by mattmoor** [comment]
Walking thru failures:
 * Tekton - all legs: failures applying patch they are carrying
 * Eventing: downstream has a problem with a breaking `trace` change: https://github.com/knative/pkg/runs/4661167676?check_suite_focus=true#step:7:491
 * Eventing Kafka (both): has same problem as above: https://github.com/knative/pkg/runs/4661168049?check_suite_focus=true#step:7:336
 * Eventing KEDA: downstream has K8s pinned https://github.com/knative-sandbox/eventing-autoscaler-keda/blob/04b969d0a9afcd2f9a73ff0aa5a7326130323918/go.mod#L23-L30
 * Eventing Redis: downstream has an OTel conflict: https://github.com/knative/pkg/runs/4661168191?check_suite_focus=true#step:7:160

I'll TAL at what we need to fix Eventing, and I already started a thread in Tekton slack to pre-flight bumping their dependency to fix this.

**Comment by mattmoor** [comment]
From the OTel CHANGELOG:
> The concept of a remote and local Span stored in a context is unified to just the current Span. Because of this "go.opentelemetry.io/otel/trace".RemoteSpanContextFromContext is removed as it is no longer needed. Instead, "go.opentelemetry.io/otel/trace".SpanContextFromContex can be used to return the current Span. If needed, that Span's SpanContext.IsRemote() can then be used to determine if it is remote or not. (#1731)

**Comment by mattmoor** [comment]
The eventing issue seems resolved, so I expect the kafka repos to be unblocked once their dependency on eventing is bumped (both have 9+ day old PRs with e2e failures/flakes)

**Comment by mattmoor** [comment]
Ok, so KEDA is fixed by: https://github.com/knative-sandbox/eventing-autoscaler-keda/pull/223

This leaves the 2x Kafka failures, 3x Tekton failures, and the codecov leg, which always seems to fail.

I'm inclined to try to get this merged...

**Comment by nak3** [comment]
LGTM :tada:

**Comment by mattmoor** [comment]
cc @n3wscott for codegen changes

File: go.mod [hunk 1]
```diff
@@ -41,11 +44,11 @@ require (
 	google.golang.org/grpc v1.42.0
 	google.golang.org/protobuf v1.27.1
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
-	k8s.io/api v0.21.4
-	k8s.io/apiextensions-apiserver v0.21.4
-	k8s.io/apimachinery v0.21.4
-	k8s.io/client-go v0.21.4
-	k8s.io/code-generator v0.21.4
+	k8s.io/api v0.22.2
```
Comment (dprotaso) [inline go.mod:47]: probably want to bump to the latest patch now `v0.22.5`

**Review by dprotaso — CHANGES_REQUESTED** [review]
we should bump to the latest patch which is `0.22.5`

**Comment by mattmoor** [comment]
@dprotaso done.  I'm going to stage a PR to tekton/pipelines to confirm we'll be good there, but this is RFAL.

**Pull request merged**